js/home.js

```diff
@@ -175,3 +298,148 @@ function addFavoriteButtonEvent () {
     })
   })
 }
+
+function showModal (id, event) {
```

Comment:
Works perfect.  I do agree about refactoring to species though as it would be much more clear.  I'm going to try to fix that favorites bug first (great catch).